src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -84,10 +86,30 @@ ov::npuw::DeviceProperties get_properties_per_device(const std::shared_ptr<const
 }  // namespace npuw
 }  // namespace ov
 
+std::shared_ptr<ov::npuw::ICompiledModel>
+ov::npuw::CompiledModelFactory::create(const std::shared_ptr<ov::Model>& model,
+                                       const std::shared_ptr<const ov::IPlugin>& plugin,
+                                       const ov::AnyMap& properties) {
+    LOG_DEBUG("CompiledModelFactory::create");
+    LOG_BLOCK();
+    std::shared_ptr<ov::npuw::ICompiledModel> compiled_model;
+    auto use_dynamic_llm_key = ov::intel_npu::npuw::dynamic_llm::enabled.name();
+    if (properties.count(use_dynamic_llm_key) &&
+        properties.at(use_dynamic_llm_key).as<bool>() == true) {
+        LOG_DEBUG("ov::npuw::LLMCompiledModel will be created.");
+        compiled_model = std::make_shared<ov::npuw::LLMCompiledModel>(model, plugin, properties);
+    } else {
+        LOG_DEBUG("ov::npuw::CompiledModel will be created.");
+        compiled_model = std::make_shared<ov::npuw::CompiledModel>(model, plugin, properties);
+    }
+    LOG_DEBUG("Done");
+    return compiled_model;
+}
+
 ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
                                        const std::shared_ptr<const ov::IPlugin>& plugin,
                                        const ov::AnyMap& properties)
-    : ov::ICompiledModel(model, plugin),
+    : ov::npuw::ICompiledModel(model, plugin),
```

Comment:
?

Reply:
Will fix this!

Reply:
There's nothing to fix hence it is a question

Reply:
Removed usage of common interface here

Reply:
Returned back